pkg/pod/ttlDestroyHandler_test.go

```diff
@@ -0,0 +1,59 @@
+package pod
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/dhis2-sre/rabbitmq/pgk/queue"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type mockQueueProducer struct{ mock.Mock }
+
+func (m *mockQueueProducer) Produce(channel queue.Channel, payload any) {
+	m.Called(channel, payload)
+}
+
+func Test_ttlDestroyHandler_Handle(t *testing.T) {
+	producer := &mockQueueProducer{}
+	handler := NewTTLDestroyHandler(producer)
+	pod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"im-id":                 "1",
+				"im-creation-timestamp": strconv.Itoa(int(time.Now().Add(time.Minute * 5).Unix())),
+				"im-ttl":                "300",
+			},
+		},
+	}
+
+	err := handler.Handle(pod)
+
+	require.NoError(t, err)
+	producer.AssertExpectations(t)
+}
+
+func Test_ttlDestroyHandler_Handle_Destroy(t *testing.T) {
+	producer := &mockQueueProducer{}
+	var channel queue.Channel = "ttl-destroy"
+	producer.On("Produce", channel, struct{ ID uint }{ID: 1})
+	handler := NewTTLDestroyHandler(producer)
+	pod := v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"im-id":                 "1",
+				"im-creation-timestamp": strconv.Itoa(int(time.Now().Add(time.Minute * -5).Unix())),
```

Comment:
I would use https://pkg.go.dev/time#Time.Sub. Makes it more clear that it was created in the past IMHO 😄 

If I am not mistaken "im-creation-timestamp"+"im-ttl" amount to now. I would avoid this case as this could become a flaky test. Might be unlikely since we are using seconds precision, but why even bother.

Reply:
But Time.Sub takes Time as argument and we need Duration.

 "im-creation-timestamp"+"im-ttl" == now - 5 minutes thus resulting in a call to the producer requesting the instance being destroyed.